structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py

```diff
@@ -148,671 +183,537 @@ def _check_env_temp(T: float) -> None:
+            raise ValueError(
+                f"Curing time {sum(dt)} and time of loading {t0} do not"
+                " match."
+            )
+    return np.sum(dt * np.exp(13.65 - (4000 / (273 + T_cur))))
+
+
+def t0_adj(_tT: float, cem_class: str) -> float:
     """Calculate the modified age at loading (t0) to account for the
         effect of the type of cement and curing temperature on the
         degree of hydration and - in turn - on creep.
 
     Defined in fib Model Code 2010 (2013), Eq. 5.1-73.
 
-    args:
-        t0 (float): The age of the concrete in days at which the
-            loading is applied.
-        T_cur (float): The temperature of the environment during curing
-            in degrees Celcius.
-        TABULAR_VALUES (dict): Material constants for different cement
-            types and aggregate types as given by the fib Model Code
-            2010.
-    returns:
+    Args:
+        _tT (float): Temperature adjusted concrete age in days.
+        cem_class (str): The cement strength class that is used.
+            The choices are:
+                '32.5 N',
+                '32.5 R', '42.5 N',
+                '42.5 R', '52.5 N', '52.5 R'.
+    Returns:
         float: The temperature corrected age of the concrete in days at
             loading, accounting for the effect of the cement type.
             For slow hardening concrete, the creep coefficient is
             increased due to the lower modified age at loading.
     """
-    _check_age_at_loading(t0)
-    t0T = _calc_temp_corr_age(t0, T_cur)
-    ALPHA = TABULAR_VALUES['alpha']
-    return max(t0T * ((9 / (2 + t0T**1.2)) + 1) ** ALPHA, 0.5)
+    return max(_tT * ((9 / (2 + _tT**1.2)) + 1) ** _alpha[cem_class], 0.5)
 
 
-def _calc_notional_drying_shrinkage(fcm: float, TABULAR_VALUES: dict) -> float:
+def eps_cds0(fcm: float, cem_class: str) -> float:
     """Calculate the notional drying shrinkage.
 
     Defined in fib Model Code 2010 (2013), Eq. 5.1-80.
 
-    args:
+    Args:
         fcm (float): The mean compressive strength of the concrete in
             MPa.
-        TABULAR_VALUES (dict): Material constants for different cement
-            types and aggregate types as given by the fib Model Code
-            2010.
+        cem_class (str): The cement strength class that is used.
+            The choices are:
+                '32.5 N',
+                '32.5 R', '42.5 N',
+                '42.5 R', '52.5 N', '52.5 R'.
 
-    returns:
+    Returns:
         float: The notional drying shrinkage in mm/mm.
     """
     return (
-        (220 + 110 * TABULAR_VALUES['alpha_ds1'])
-        * np.exp(-TABULAR_VALUES['alpha_ds2'] * fcm)
-    ) * 1e-6
+        (220 + 110 * _alpha_ds1[cem_class])
+        * np.exp(-_alpha_ds2[cem_class] * fcm)
+        * 1e-6
+    )
 
 
-def _calc_beta_ds(
-    time: np.ndarray, ts: float, notional_size: float
-) -> np.ndarray:
+def beta_ds(time: np.ndarray, ts: float, notional_size: float) -> np.ndarray:
```

Comment:
Maybe we should use ArrayLike? so that the user can input an array, or a list, or anything that can be converted to array?

Reply:
Done